src/impls.rs

```diff
@@ -47,32 +48,44 @@ impl<T: Config> Pallet<T> {
 				<DisputeInfo<T>>::contains_key(dispute_id),
 			Error::<T>::InvalidDispute
 		);
-		<VoteInfo<T>>::try_mutate(dispute_id, vote_round, |maybe| match maybe {
-			None => Err(Error::<T>::InvalidVote),
-			Some(vote) => {
-				// Ensure vote has not already been executed, and vote must be tallied
-				ensure!(!vote.executed, Error::<T>::VoteAlreadyExecuted);
-				ensure!(vote.tally_date > 0, Error::<T>::VoteNotTallied);
-				// Ensure vote must be final vote and that time has to be pass after the vote is tallied
-				ensure!(
-					<VoteRounds<T>>::get(vote.identifier) == vote.vote_round,
-					Error::VoteNotFinal
-				);
-				ensure!(
-					Self::now().saturating_sub(vote.tally_date) >= 1 * DAYS,
-					Error::<T>::TallyDisputePeriodActive
-				);
-				vote.executed = true;
-				let dispute =
-					<DisputeInfo<T>>::get(dispute_id).ok_or(Error::<T>::InvalidDispute)?;
-				<OpenDisputesOnId<T>>::mutate(dispute.query_id, |maybe| {
-					if let Some(disputes) = maybe {
-						disputes.saturating_dec();
-					}
-				});
-				vote.result = Some(result);
-				Ok(())
-			},
+		<VoteInfo<T>>::try_mutate(dispute_id, vote_round, |maybe| -> DispatchResult {
+			match maybe {
+				None => Err(Error::<T>::InvalidVote.into()),
+				Some(vote) => {
+					// Ensure vote has not already been executed, and vote must be tallied
+					ensure!(!vote.executed, Error::<T>::VoteAlreadyExecuted);
+					ensure!(vote.tally_date > 0, Error::<T>::VoteNotTallied);
+					// Ensure vote must be final vote and that time has to be pass after the vote is tallied
+					ensure!(
+						<VoteRounds<T>>::get(vote.identifier) == vote.vote_round,
+						Error::<T>::VoteNotFinal
+					);
+					ensure!(
+						Self::now().saturating_sub(vote.tally_date) >= 1 * DAYS,
+						Error::<T>::TallyDisputePeriodActive
+					);
+					vote.executed = true;
+					let dispute =
+						<DisputeInfo<T>>::get(dispute_id).ok_or(Error::<T>::InvalidDispute)?;
+					<OpenDisputesOnId<T>>::mutate(dispute.query_id, |maybe| {
+						if let Some(disputes) = maybe {
+							disputes.saturating_dec();
+						}
+					});
+					vote.result = Some(result);
+
+					// handling transfer of dispute fee
+					let dispute_account =
+						&T::PalletId::get().into_sub_account_truncating(DISPUTE_SUB_ACCOUNT_ID);
+					let dest = match result {
+						VoteResult::Passed | VoteResult::Invalid => &vote.initiator,
+						VoteResult::Failed => &dispute.disputed_reporter,
+					};
+					T::Token::transfer(dispute_account, dest, vote.fee, false)?;
```

Comment:
Does this function only transfer the dispute fee for a single round, or for each round of voting? It looks like just a single round, but I could be missing it because I'm not super familiar with Rust. It should transfer it for each round of voting like [here](https://github.com/tellor-io/governance/blob/main/contracts/Governance.sol#L207).

Bumping [this issue](https://github.com/tellor-io/tellor-parachain-contracts/issues/14) as well. And the current `executeVote` logic on the contracts side [here](https://github.com/tellor-io/tellor-parachain-contracts/blob/main/src/ParachainGovernance.sol#L347).

Reply:
Hi @oraclown,

As discussed, we'll be addressing this in the new issue, so is there anything else you want to highlight here?
Thanks

Reply:
@pawanbisht62 We still need to add the iteration of the vote rounds in this PR to conclude please. The merging of the relevant dispatchable functions will be tackled in another PR and i will create an issue for it now.

Reply:
Nothing else from me. Thanks, @pawanbisht62 

I made that issue (which might need more detail) #36,  @evilrobot-01 . I didn't see an existing one, but feel free to close as duplicate if I missed it